torch/autograd/variable.py

```diff
@@ -868,7 +875,7 @@ def __len__(self):
         return len(self.data)
 
     def __iter__(self):
-        return iter(map(lambda i: self[i], range(self.size(0))))
+        return iter(imap(lambda i: self[i], range(self.size(0))))
```

Comment:
Why + did it fix it now?

Reply:
I misdiagnosed the issue previously; it turns out the old implementation of zip does interleave accesses as necessary. The real problem is that the map in iter eagerly performs the index on all elements (since it materializes a list.) Making it produce a generator finally fixes the problem.